envy/lib/config/schema.py

```diff
@@ -146,7 +146,13 @@ def __validate_environment_setup_steps(environment: {}) -> bool:
             }
         ],
         Optional("services", default={}): {Optional("compose-file"): str},
-        Optional("network", default=None): str,
+        Optional("network", default={}): Or(
+            "host",
+            {
+                Optional("name", default=None): str,
+                Optional("ports", default=None): {Or(int, str): Or(int, str)},
```

Comment:
I think I'd rather this be a list of pairs like docker-compose does it, rather than this dict structure.
So that the envyfile looks like this:
```
network:
  ports:
    - 9091
    - 8080:80
```

With the current implementation we'd have to write `9091: 9091` right?

Reply:
Yeah, I just picked the easiest option to match with the python docker api, so I'm fine with this change, but it does bring up at least one question.
What should the result of ` - 9091` be? For example, in docker-compose, it maps 9091 in the container to an ephemeral host port, but I don't think we want to be doing that ourselves and I don't think the python library accepts that format.

Reply:
Oh, NVM, the api *does* support this.

Reply:
Ah you're right, I thought `- 9091` mapped 9091 in the container to 9091 on the host but it seems to default to that if available and otherwise uses a random ephemeral.
I think the result of `- 9091` in envy should be `9091:9091` passed to docker.

Reply:
I don't think we should deviate from Docker standards unless we have a very good reason to do so. I know that we are supposed to be transparent about what our backend is, but if there is common knowledge we can build on and no reason not to we should use it.

Reply:
Fair point, I agree with that!